railties/test/application/configuration_test.rb

```diff
@@ -78,6 +78,18 @@ def restore_default_config
       end
     end
 
+    test "Rails.env falls back to development if RAILS_ENV == '' and RACK_ENV == nil" do
```

Comment:
With tests I think it's usually better to write out what you're expecting versus writing code. So in that case this would be "Rails.env falls back to development if RAILS_ENV is blank and RACK_ENV is nil"